DigitalLearningSolutions.Web/Controllers/LoginController.cs

```diff
@@ -24,54 +35,107 @@ public IActionResult Index()
         [HttpPost]
         public IActionResult Index(LoginViewModel model)
         {
-            // TODO: HEEDLS-364 - Overwrite this old code for automatic sign in with new code that signs in the user after validation
-            if (model.Username?.ToLower() == "testusernameerror")
+            if (!ModelState.IsValid)
             {
-                ModelState.AddModelError("username", "There is a username error.");
+                return View("Index", model);
             }
 
-            if (model.Username?.ToLower() == "testpassworderror")
+            AdminUser? adminUser = userService.GetAdminUserByUsername(model.Username);
+            List<DelegateUser> delegateUsers = userService.GetDelegateUsersByUsername(model.Username);
+
+            if (adminUser == null && delegateUsers.Count == 0)
             {
-                ModelState.AddModelError("password", "There is a password error.");
+                ModelState.AddModelError("Username","No account with that email address or user ID could be found.");

```

Comment:
I can see a cheeky no space after "," 😄 Might be worth running the formatter quickly!